src/org/traccar/protocol/Jt600ProtocolDecoder.java

```diff
@@ -256,11 +259,29 @@ private Position decodeU01(String sentence, Channel channel, SocketAddress remot
         position.set(Position.KEY_CID, parser.nextInt());
         position.set(Position.KEY_LAC, parser.nextInt());
         position.set(Position.KEY_GSM, parser.nextInt());
-        position.set(Position.KEY_ODOMETER, parser.nextLong() * 1000);
+        position.set(Position.KEY_ODOMETER, parser.nextLong() * 100);
         position.set(Position.KEY_INDEX, parser.nextInt());
 
-        return position;
+        switch (messageType) {
+            case "U01":
+            case "U02":
+            case "U03":
+                String checkSum = parser.next();
+                int calculatedCheckSum = checkSum(sentence.substring(1, sentence.length() - checkSum.length() - 1));
+                if (Integer.parseInt(checkSum, 16) == calculatedCheckSum) {
+                    sendResponse(channel, "(S39)");
+                    return position;
+                } else {
+                    return null;
+                }
+            case "U06":
+                sendResponse(channel, "(S20)");
+                return position;
+            default:
+                return null;
```

Comment:
Documentation describes only these types of messages. Since there is no description on other types, like U07, I consider them invalid. Again, other types of messages may require a special type of response. Otherwise, device repeats such messages and floods the server. Thus, I believe it's better to handle only known types of messages.